src/nxt_isolation.c

```diff
@@ -197,6 +211,42 @@ nxt_isolation_set(nxt_task_t *task, nxt_conf_value_t *isolation,
 }
 
 
+#if (NXT_HAVE_CGROUP)
+
+static nxt_int_t
+nxt_isolation_set_cgroup(nxt_task_t *task, nxt_conf_value_t *isolation,
+    nxt_process_t *process)
+{
+    nxt_str_t         str;
+    nxt_conf_value_t  *obj;
+
+    static nxt_str_t  cgname = nxt_string("cgroup");
+    static nxt_str_t  path = nxt_string("path");
+
+    obj = nxt_conf_get_object_member(isolation, &cgname, NULL);
+    if (obj == NULL) {
+        return NXT_OK;
+    }
+
+    obj = nxt_conf_get_object_member(obj, &path, NULL);
+    if (obj == NULL) {
+        return NXT_ERROR;
+    }
+
+    nxt_conf_get_string(obj, &str);
+    process->isolation.cgroup.path = nxt_mp_alloc(process->mem_pool,
+                                                  str.length + 1);
+    nxt_memcpy(process->isolation.cgroup.path, str.start, str.length);
+    process->isolation.cgroup.path[str.length] = '\0';
```

Comment:
Do you think we should add a function that transforms a (not-NUL-terminated) buffer into a (NUL-terminated) string?  I'm guessing this is something we will encounter often in this code base.

I have a pull request for shadow-utils doing this, and it simplifies and adds safety to the code.

See:

<https://github.com/shadow-maint/shadow/pull/569/files#diff-12b560bab6b4fb8f7f3a16f01aaa994de539a8bed3058c976be0daebe16405c1>

Reply:
Probably, there are at least two other places in that file doing the same.